python-lib/config.py

```diff
@@ -82,19 +82,32 @@ def get_hdfs_location(dataset_config: Mapping[AnyStr, Any], sf_stage_name: AnySt
     return f'{sf_stage_name}{path}/'
 
 
-def get_snowflake_to_hdfs_query(sf_location: AnyStr, sf_table_name: AnyStr) -> AnyStr:
+def get_snowflake_to_hdfs_query(sf_location: AnyStr, sf_table_name: AnyStr,
+                                sf_schema: List[Mapping[AnyStr, AnyStr]]) -> AnyStr:
     """
     Gets a COPY statement for copying from a Snowflake Table to an HDFS location
-    :param sf_location:
-    :param sf_table_name:
+    :param sf_location: HDFS location
+    :param sf_table_name: Snowflake table
+    :param sf_schema: Snowflake schema
     :return: a SQL COPY statement
     """
     # moving this function here isn't strictly necessary as it's not re-used, but it makes
     # things significantly easier to unit test.
 
+    # Generate SELECT clause and cast TIMESTAMP_TZ, TIMESTAMP_LTZ to TIMESTAMP
+    # This is required as the COPY command does not support TZ and LTZ
+    columns = [
+        f'"{c["name"]}"'
+        + (f'::TIMESTAMP AS "{c["name"]}"' if c['originalType'] in ['TIMESTAMPLTZ', 'TIMESTAMPTZ'] else '')
```

Comment:
```suggestion
        + (f'::TIMESTAMP_NTZ AS "{c["name"]}"' if c['originalType'] in ['TIMESTAMPLTZ', 'TIMESTAMPTZ'] else '')
```

Would this address the potential differences in `TIMESTAMP_TYPE_MAPPING`?

Reply:
Yes - this will resolve the issue! I was thinking about same solution.